[libpas] Log libpas configuration when LibpasMallocReportConfig=1

Achierius opened the pull request:
#### 162e30c5ee9f92440b781a57d6aaba663cc90fa5
<pre>
[libpas] Log libpas configuration when LibpasMallocReportConfig=1
<a href="https://bugs.webkit.org/show_bug.cgi?id=302473">https://bugs.webkit.org/show_bug.cgi?id=302473</a>
<a href="https://rdar.apple.com/164639889">rdar://164639889</a>

Reviewed by Yijia Huang.

When LibpasMallocReportConfig=1 is set, libpas will dump
a summary of its main configurations (MTE, scavenger timings,
object-size bounds, runtime-heap-configs, etc.) to stderr.
The dump will currently look something like the following:
```
jsc(76428,0x103) malloc: libpas config:
        Deallocation Log (Max Entries, Max Bytes): 100, 50000B
        Scavenger (Period, Deep-Sleep Timeout, Epoch-Delta): 100.00ms, 10000.00ms, 600000000
        MTE (Enabled/Mode-Bits/Tagging-Rate/Medium-Enabled/Lockdown/Hardened): (0, 0, 0, 0, 0, 0)
        Using System Heap: 0
        Heap-Config bmalloc:
                Page Configs (Enabled/MTE Taggable, Static Max Obj Size):
                        Small Segregated: 1/1, 1616B
                        Medium Segregated: 1/1, 12800B
                        Small Bitfit: 1/1, 1600B
                        Medium Bitfit : 1/1, 12800B
                        Marge Bitfit : 1/0, 417792B
                Runtime Heap Config Size-Maximums (Segregated, Bitfit, Baseline Dir, No-View-Cache Dir):
                        Flex: 4294967295B, 0B, 0B, 4294967295B
                        Intrinsic: 4294967295B, 4294967295B, 0B, 0B
                        Typed: 4294967295B, 0B, 0B, 4294967295B
                        Primitive: 4294967295B, 4294967295B, 0B, 4294967295B
        Heap-Config jit:
                Page Configs (Enabled/MTE Taggable, Static Max Obj Size):
                        Small Segregated: 1/0, 1632B
                        Medium Segregated: 0/0, 0B
                        Small Bitfit: 1/0, 1012B
                        Medium Bitfit : 1/0, 64768B
                        Marge Bitfit : 0/0, 0B
                Runtime Heap Config Size-Maximums (Segregated, Bitfit, Baseline Dir, No-View-Cache Dir):
                        Flex: 2000B, 4294967295B, 0B, 0B
        Heap-Config utility:
                Page Configs (Enabled/MTE Taggable, Static Max Obj Size):
                        Small Segregated: 1/0, 1400B
                        Medium Segregated: 0/0, 0B
                        Small Bitfit: 0/0, 0B
                        Medium Bitfit : 0/0, 0B
                        Marge Bitfit : 0/0, 0B
```
Along with some ifdefs to enable logging for the iso-heap-config in the
case that that&apos;s actually compiled in (which currently we do not do).

Canonical link: <a href="https://commits.webkit.org/303347@main">https://commits.webkit.org/303347@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/12aaff6549eec486c08519a5201795e7e4f6c89a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132093 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4585 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43113 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139608 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/84007 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/4d11892e-f3a0-400a-ae67-07e49c59640c) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4533 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4347 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100973 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/84007 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135039 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3215 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118309 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81764 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3112 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/1003 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82827 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/124156 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111886 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36430 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142254 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/130600 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4255 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37008 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109346 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4336 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/3704 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109520 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27745 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3231 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114584 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57499 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4309 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32978 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/163567 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4140 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/42513 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4400 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4268 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

hyjorc1 commented on Source/bmalloc/libpas/src/libpas/pas_mte_config.c:
```diff
@@ -31,20 +31,44 @@
+#if PAS_ENABLE_ISO
+extern const pas_heap_config iso_heap_config;
+#endif // PAS_ENABLE_ISO
+extern const pas_heap_config pas_utility_heap_config;
```
Dup of line 71?

Achierius replied:
Will remove

hyjorc1 reviewed (approved):
r=me if EWS bots are happy.

Achierius committed:
[libpas] Log libpas configuration when LibpasMallocReportConfig=1
https://bugs.webkit.org/show_bug.cgi?id=302473
rdar://164639889

Reviewed by Yijia Huang.

When LibpasMallocReportConfig=1 is set, libpas will dump
a summary of its main configurations (MTE, scavenger timings,
object-size bounds, runtime-heap-configs, etc.) to stderr.
The dump will currently look something like the following:
```
jsc(76428,0x103) malloc: libpas config:
        Deallocation Log (Max Entries, Max Bytes): 100, 50000B
        Scavenger (Period, Deep-Sleep Timeout, Epoch-Delta): 100.00ms, 10000.00ms, 600000000
        MTE (Enabled/Mode-Bits/Tagging-Rate/Medium-Enabled/Lockdown/Hardened): (0, 0, 0, 0, 0, 0)
        Using System Heap: 0
        Heap-Config bmalloc:
                Page Configs (Enabled/MTE Taggable, Static Max Obj Size):
                        Small Segregated: 1/1, 1616B
                        Medium Segregated: 1/1, 12800B
                        Small Bitfit: 1/1, 1600B
                        Medium Bitfit : 1/1, 12800B
                        Marge Bitfit : 1/0, 417792B
                Runtime Heap Config Size-Maximums (Segregated, Bitfit, Baseline Dir, No-View-Cache Dir):
                        Flex: 4294967295B, 0B, 0B, 4294967295B
                        Intrinsic: 4294967295B, 4294967295B, 0B, 0B
                        Typed: 4294967295B, 0B, 0B, 4294967295B
                        Primitive: 4294967295B, 4294967295B, 0B, 4294967295B
        Heap-Config jit:
                Page Configs (Enabled/MTE Taggable, Static Max Obj Size):
                        Small Segregated: 1/0, 1632B
                        Medium Segregated: 0/0, 0B
                        Small Bitfit: 1/0, 1012B
                        Medium Bitfit : 1/0, 64768B
                        Marge Bitfit : 0/0, 0B
                Runtime Heap Config Size-Maximums (Segregated, Bitfit, Baseline Dir, No-View-Cache Dir):
                        Flex: 2000B, 4294967295B, 0B, 0B
        Heap-Config utility:
                Page Configs (Enabled/MTE Taggable, Static Max Obj Size):
                        Small Segregated: 1/0, 1400B
                        Medium Segregated: 0/0, 0B
                        Small Bitfit: 0/0, 0B
                        Medium Bitfit : 0/0, 0B
                        Marge Bitfit : 0/0, 0B
```
Along with some ifdefs to enable logging for the iso-heap-config in the
case that that's actually compiled in (which currently we do not do).

Canonical link: https://commits.webkit.org/303347@main
